src/main/bootstrap/mod.rs

```diff
@@ -336,13 +336,13 @@ fn shuffled_bootstrap_peers(
 
 /// Appends peers received from `ServiceDiscMeta` to the peers list
 fn receive_peers_into(peers: &mut Vec<PeerInfo>, rx: Receiver<HashSet<PeerInfo>>) {
-    let discovered_peers: HashSet<PeerInfo> = {
-        let our_peers: HashSet<&PeerInfo> = HashSet::from_iter(peers.iter());
-        rx.try_iter()
-            .flatten()
-            .filter(|peer| !our_peers.contains(peer))
-            .collect()
-    };
+    let our_peers: HashSet<&PeerInfo> = HashSet::from_iter(peers.iter());
```

Comment:
Do we really need to specify `our_peers` type?

Reply:
If I omit the type, this is what happens:

```rust
error[E0282]: type annotations needed
   --> src/main/bootstrap/mod.rs:339:21
    |
339 |     let our_peers = HashSet::from_iter(peers.iter());
    |         ---------   ^^^^^^^^^^^^^^^^^^ cannot infer type for `S`
    |         |
    |         consider giving `our_peers` a type

```
So I guess there has to be a type annotation for `our_peers`